desktop/cmp/treemap/SplitTreeMapModel.js

```diff
@@ -0,0 +1,85 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2019 Extremely Heavy Industries Inc.
+ */
+import {HoistModel, managed} from '@xh/hoist/core';
+import {bindable} from '@xh/hoist/mobx';
+import {throwIf, withDefault} from '@xh/hoist/utils/js';
+
+import {TreeMapModel} from './TreeMapModel';
+
+/**
+ * Core Model for a SplitTreeMap.
+ *
+ * Binds to a Store (or GridModel) and splits the data into two managed child TreeMaps.
+ * Users should specify a `mapFilter` function to control how records are divided
+ * across the two TreeMaps.
+ *
+ * Additionally, accepts and passes along all settings for TreeMapModel.
+ * @see TreeMapModel
+ */
+@HoistModel
+export class SplitTreeMapModel {
+
+    //------------------------
+    // Immutable public properties
+    //------------------------
+    /** @member {function} */
+    mapFilter;
+    /** @member {function} */
+    mapTitleFn;
+    /** @member {string} */
+    orientation;
+
+    /** @member {TreeMapModel} */
+    @managed primaryMapModel;
+    /** @member {TreeMapModel} */
+    @managed secondaryMapModel;
+
+    //------------------------
+    // Observable API
+    //------------------------
+    /** @member {Object} */
+    @bindable.ref highchartsConfig = {};
+
+    /**
+     * @param {Object} c - SplitTreeMapModel configuration.
+     * @param {function} c.mapFilter - A filter function used when processing data. Receives (record), returns boolean.
+     *      Records that pass the filter will be placed into the primary TreeMap, and the rest into the secondary TreeMap.
+     *      If not passed, will default to: { return record.valueField >= 0; }
+     * @param {function} [c.mapTitleFn] - Function to render map titles. Receives map name ['primary', 'secondary'] and region TreeMapModel.
```

Comment:
Will this be more convenient if we passed in an `isPrimary` boolean for the first argument?  I think so - would avoid the need to check for that string, and I think the nature of SplitTreeMap means we're gonna keep having two maps so a boolean would work.

Don't feel that strongly about it, but wanted to get that suggestion in.

Reply:
We like this suggestion and will make this change now.